src/app/@core/interfaces/exercise.interface.ts

```diff
@@ -0,0 +1,12 @@
+import { Difficulty, Region, WeightType } from 'src/app/pages/constans';
+
+export interface ExerciseInterface {
```

Comment:
id and userId should be required.
use different interfaces for getting and creating exercises 
read about Pick and Omit types in TypeScript